### PR TITLE
Upgrade to cosmos-sdk v0.45.6 and ibc-go v4.0.0-rc2

### PR DESCRIPTION
**Pull request by devashishdxt**

### CODE REVIEW

**Review by crodriguezvega — COMMENTED** [review]
Thanks, @devashishdxt, for doing this upgrade!

We need to bump the go version in go.mod to 1.18 and also here in the [Dockerfile](https://github.com/cosmos/interchain-accounts-demo/blob/master/Dockerfile#L1). Would you mind doing that so that CI passes? Thanks!

**Comment by chatton** [comment]
Hi @devashishdxt, we don't need to worry about the docker build failure here. We need to update the action to not run on forks.

**Review by crodriguezvega — APPROVED** [review]
Thanks for doing this work, @devashishdxt, and sorry that it took a while to review the PR. We'll try to review and merge it as soon as possible. I just left a comment, but other than that I think it looks good. Thanks again!

As potential addition, maybe you could update to v4.0.0-rc2, if you want. Otherwise, we will do that in a follow PR or update when the final v4.0.0 comes out.

**Comment by devashishdxt** [comment]
> As potential addition, maybe you could update to v4.0.0-rc2, if you want. Otherwise, we will do that in a follow PR or update when the final v4.0.0 comes out.

Done

File: x/inter-tx/ibc_module.go [hunk 1]
```diff
@@ -39,8 +42,17 @@ func (im IBCModule) OnChanOpenInit(
 	chanCap *capabilitytypes.Capability,
 	counterparty channeltypes.Counterparty,
 	version string,
-) error {
-	return im.keeper.ClaimCapability(ctx, chanCap, host.ChannelCapabilityPath(portID, channelID))
+) (string, error) {
+	if strings.TrimSpace(version) == "" {
```
[the remark this answers is not in the document]
Comment (devashishdxt) [reply x/inter-tx/ibc_module.go]: Done

**Review by seantking — APPROVED** [review]
:handshake:

File: proto/intertx/tx.proto [hunk 2]
```diff
@@ -22,6 +22,7 @@ message MsgRegisterAccount {
 
   string owner                      = 1;
   string connection_id              = 2 [(gogoproto.moretags) = "yaml:\"connection_id\""];
+  string version                    = 3 [(gogoproto.moretags) = "yaml:\"version\""];
```
Comment (damiannolan) [inline proto/intertx/tx.proto:25]: I think you can omit the yaml tags for this field! :)

File: x/inter-tx/ibc_module.go [hunk 3]
```diff
@@ -103,7 +108,7 @@ func (im IBCModule) OnRecvPacket(
 	packet channeltypes.Packet,
 	relayer sdk.AccAddress,
 ) ibcexported.Acknowledgement {
-	return channeltypes.NewErrorAcknowledgement("cannot receive packet via interchain accounts authentication module")
+	return channeltypes.NewErrorAcknowledgement(sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot receive packet via interchain accounts authentication module"))
```
Comment (damiannolan) [inline x/inter-tx/ibc_module.go:111]: maybe `ErrInvalidRequest` is more fitting? 

```suggestion
	return channeltypes.NewErrorAcknowledgement(sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "cannot receive packet via interchain accounts authentication module"))
```

**Review by damiannolan — APPROVED** [review]
Thanks ❤️

File: proto/intertx/tx.proto (hunk 2 above)
Comment (devashishdxt) [reply proto/intertx/tx.proto]: Done

**Pull request merged**